processor/batchprocessor/config.go

```diff
@@ -1,68 +1,69 @@
-// Copyright The OpenTelemetry Authors
-// SPDX-License-Identifier: Apache-2.0
+// Code generated by github.com/atombender/go-jsonschema, DO NOT EDIT.
 
-package batchprocessor // import "go.opentelemetry.io/collector/processor/batchprocessor"
+package batchprocessor
 
-import (
-	"errors"
-	"fmt"
-	"strings"
-	"time"
+import "encoding/json"
+import "fmt"
+import "time"
 
-	"go.opentelemetry.io/collector/component"
-)
-
-// Config defines configuration for batch processor.
+// Configuration parameters for the batch processor.
 type Config struct {
-	// Timeout sets the time after which a batch will be sent regardless of size.
-	// When this is set to zero, batched data will be sent immediately.
-	Timeout time.Duration `mapstructure:"timeout"`
+	// MetadataCardinalityLimit corresponds to the JSON schema field
+	// "metadata_cardinality_limit".
+	MetadataCardinalityLimit int `mapstructure:"metadata_cardinality_limit,omitempty"`
 
-	// SendBatchSize is the size of a batch which after hit, will trigger it to be sent.
-	// When this is set to zero, the batch size is ignored and data will be sent immediately
-	// subject to only send_batch_max_size.
-	SendBatchSize uint32 `mapstructure:"send_batch_size"`
+	// MetadataKeys corresponds to the JSON schema field "metadata_keys".
+	MetadataKeys []string `mapstructure:"metadata_keys,omitempty"`
 
-	// SendBatchMaxSize is the maximum size of a batch. It must be larger than SendBatchSize.
-	// Larger batches are split into smaller units.
-	// Default value is 0, that means no maximum size.
-	SendBatchMaxSize uint32 `mapstructure:"send_batch_max_size"`
+	// SendBatchMaxSize corresponds to the JSON schema field "send_batch_max_size".
+	SendBatchMaxSize int `mapstructure:"send_batch_max_size,omitempty"`
```

Comment:
Do we care much that the `uint` is now an `int`? 
I suppose we don't, because we can add a check to make sure it's not negative. Also, I doubt that the integers people configure are so large that an `int` wouldn't fit them.

Reply:
We do - at the very least we cannot change types like this. We need to deprecate and move over with a set of changes over multiple releases.

Reply:
I could update my fork of `go-jsonschema` to use uint whenever there is a `minimum: 0` or an `exclusiveMinimum: 0`? It seems like a reasonable feature - maybe it'll also get accepted upstream.